### PR TITLE
Add install requirements

### LINKED ISSUE #4: State that rehype-slug is required
### Subject of the feature

It's not documented that `rehype-slug` is required for `rehype-autolink-headings` to work.

### Problem

I'm not familiar with rehype, its dependencies and etc. so I got into this problem while using `next-mdx-enhanced` and found out that I was not alone:

https://github.com/hashicorp/next-mdx-enhanced/issues/56 (OP closed issue without comments)
https://github.com/hashicorp/next-mdx-enhanced/issues/62

It's not clear that `rehype-slug` is a required dependency. I saw the examples and thought of it just as mere examples, as my usage differs from it, eg.:

```js
module.exports = withMdxEnhanced({
  layoutPath: 'layouts',
  fileExtensions: ['mdx'],
  remarkPlugins: [],
  rehypePlugins: [require('rehype-slug'), require('rehype-autolink-headings')],
  extendFrontMatter: {
    process: (mdxContent, frontMatter) => {},
    phase: 'prebuild|loader|both',
  },
})()
```

### Expected behavior

Add a note to install instructions:

*Note that `rehype-slug` is required and should be used/invoked before `rehype-autolink-headings`

### PR DESCRIPTION
**Pull request by estevanmaito**
fix #4

What do you think about the message?

### CODE REVIEW

**Comment by wooorm** [comment]
👋

Thanks! Sorry you ran into this.

I felt like it was sort of said in the readme already, but I get that it’s vague and needs to be better.  I was thinking more like this:

```markdown
> **Note**: this plugin expects `id`s to already exist on headings.
> One way to add those automatically, is [`rehype-slug`][slug] (see example).
```

…right above `options` in the `API` section? (and `[slug]: https://github.com/rehypejs/rehype-slug` at the bottom).

This:

* Calls more attention
* In my opinion better reflects that it’s not a requirement, but a common case

What do you think?

File: readme.md [hunk 1]
```diff
@@ -221,3 +224,5 @@ abide by its terms.
 [xss]: https://en.wikipedia.org/wiki/Cross-site_scripting
 
 [sanitize]: https://github.com/rehypejs/rehype-sanitize
+
+[slug]: https://github.com/rehypejs/rehype-slug
```
Comment (wooorm) [inline readme.md:228]: Seems there’s an enter missing at the end of this line!

File: readme.md [hunk 2]
```diff
@@ -151,7 +154,7 @@ Always be wary of user input and use [`rehype-sanitize`][sanitize].
 
 ## Related
 
-*   [`rehype-slug`](https://github.com/rehypejs/rehype-slug)
+*   [`rehype-slug`][slug]
```
Comment (wooorm) [inline readme.md:157]: ✨

File: readme.md (hunk 1 above)
Comment (estevanmaito) [reply readme.md:228]: My bad! I was just editing the md and didn't install deps/run tests. Fixing...

**Comment by estevanmaito** [comment]
aaaah Windows will kill me with these line ending removals!

**Comment by wooorm** [comment]
Sweet, thanks @estevanmaito!

**Pull request merged**